server/engine/src/main/antlr4/org/eclipse/lsp/db2/parser/Db2SqlParser.g4

```diff
@@ -24,9 +24,20 @@ execRule: EXEC SQL sqlCode END_EXEC
          | {notifyError("cobolParser.missingEndExec");} EXEC SQL sqlCode DOT_FS? EOF
          | {notifyError("cobolParser.missingEndExec");} EXEC SQL;
 
-nonExecRule: result_set_locator_host_variable;
+nonExecRule: host_variable_rule;
+
+host_variable_rule: (result_set_locator_host_variable | binary_host_variable);
 
 result_set_locator_host_variable: dbs_level_01 entry_name  (USAGE IS?)? SQL TYPE IS RESULT_SET_LOCATOR VARYING;
+
+binary_host_variable: dbs_level_01 entry_name host_variable_usage binary_host_variable_type;
+binary_host_variable_type: BINARY LPARENCHAR binary_host_variable_binary_size RPARENCHAR | VARBINARY LPARENCHAR binary_host_variable_varbinary_size RPARENCHAR  | BINARY VARYING;
```

Comment:
( BINARY | VARBINARY | BINARY VARYING) LPARENCHAR binary_host_variable_varbinary_size RPARENCHAR

Reply:
Do we know the max length when using BINARY VARYING? I've kept the sizes separate in order to allow for bounds/range checking as BINARY and VARBINARY have different max sizes. (255 and 32704 respectively)

Reply:
Well theoretically, these are semantics check and not a syntax issue. But I don't mind it either ways. As a general rule I like to consolidate sub rules to avoid look ahead burden. This is explained at  [improving-the-performance-of-an-antlr-parse](https://tomassetti.me/improving-the-performance-of-an-antlr-parser/) ( check section - Consolidating sub rules)

Since this is a small rule, it doesn't matter a lot here. 

For the semantics check, we can always do these in Visitor as the binary_host_variable parser rule context would have all the appropriate value within it.